File: v1/code/go/src/jvmgo/ch07/instructions/control/goto.go
package control

import "jvmgo/ch07/instructions/base"
import "jvmgo/ch07/rtda"

// Branch always
type GOTO struct{ base.BranchInstruction }

func (self *GOTO) Execute(frame *rtda.Frame) {
	base.Branch(frame, self.Offset)
}

File: v1/code/java/jvmgo_java/ch05/src/main/java/com/github/jvmgo/instructions/control/goto_.java
package com.github.jvmgo.instructions.control;

import com.github.jvmgo.instructions.base.BranchInstruction;
import com.github.jvmgo.instructions.base.BytecodeReader;
import com.github.jvmgo.rtda.Frame;

public class goto_ extends BranchInstruction {
    @Override
    public int getOpCode() {
        return 0xa7;
    }

    @Override
    public void fetchOperands(BytecodeReader reader) throws Exception {
        this.offset = reader.read16();
    }

    @Override
    public void execute(Frame frame) throws Exception {
        branch(frame, offset);
    }
}
